### PR TITLE
Split first few sections into files; tweak to match planned TOC

### PR DESCRIPTION
**Pull request by kainino0x**
Blocking issue: this breaks `make online`.

***
<a href="https://pr-preview.s3.amazonaws.com/kainino0x/gpuweb/pull/833.html" title="Last updated on Jun 8, 2020, 6:19 PM UTC (2caf7ce)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/gpuweb/gpuweb/833/7ace193...kainino0x:2caf7ce.html" title="Last updated on Jun 8, 2020, 6:19 PM UTC (2caf7ce)">Diff</a>

### CODE REVIEW

**Comment by kainino0x** [comment]
WDYT?

File: spec/sections/010-intro.bs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+*This section is non-normative.*
```
Comment (kvark) [inline spec/sections/010-intro.bs:1]: I'm worried about starting the files with the section indices. If we decide to add section in the middle, we'd have to rename files, which is bad for history

Comment (kainino0x) [reply spec/sections/010-intro.bs:1]: History is generally pretty robust to renames. But that's why I have `010-` etc, then you can insert a section with `015-`. They're not supposed to be section numbers (though I don't know how to make that obvious).

**Comment by kainino0x** [comment]
> how do I see these changes now? everything is one giant rewrite

I should have done it in two commits. Let me just split this, it should be quick.

**Comment by kainino0x** [comment]
Done.

**Comment by kainino0x** [comment]
Downside would just be that you wouldn't be able to see the section ordering by looking at the file listing (e.g. in an editor sidebar), you'd have to look at the index.bs. Pretty minor.

**Comment by kainino0x** [comment]
I don't know of another spec that splits files in this way.

**Comment by kvark** [comment]
Ok, I think we can get away without the numbers then. It's hard to see a workflow where the ordering would be a problem.

**Comment by Kangz** [comment]
> Is there an example spec that follow a similar scheme of splitting into files?

The Vulkan spec is split into one asciidoc file per chapter.

**Comment by kainino0x** [comment]
> Ok, I think we can get away without the numbers then. It's hard to see a workflow where the ordering would be a problem.

I thought of one: it would keep the files in the correct order during code review.

**Comment by kvark** [comment]
> The Vulkan spec is split into one asciidoc file per chapter.

Right, and they [don't have them prefixed](https://github.com/KhronosGroup/Vulkan-Docs/tree/master/chapters) by numbers. It would be good to ask them if they would consider this a problem, or otherwise we can just follow the same steps.

**Comment by kainino0x** [comment]
Do we think splitting is worth it? The benefits aren't that huge, I think it just slightly improves workflow (can have multiple chapters open in my editor, and the document isn't so huge that it's easy to get lost in it) but maybe isn't worth the downsides (need multi-file-ctrl-f to find things, weird numbering or code review diffs out of order).

**Comment by kvark** [comment]
I think there is more benefit than harm in the long term in splitting the spec into a few files.

**Comment by kainino0x** [comment]
> It would be good to ask them if they would consider this a problem, or otherwise we can just follow the same steps.

I'm sure if they did they would have changed it. It would be a pretty small change.

Removed the prefixes.

**Comment by kainino0x** [comment]
Marking ready for review because I'd like to do this incrementally.

File: spec/sections/README.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Sections of this spec are split into separate files for ease of editing.
+
+Note: The numbers in front of the section filenames are for filename ordering only.
```
Comment (JusSn) [inline spec/sections/README.md:3]: This is no longer relevant.

File: spec/index.bs [hunk 3]
```diff
@@ -266,10 +240,38 @@ dictionary GPUObjectDescriptorBase {
         The initial value of {{GPUObjectBase/label|GPUObjectBase.label}}.
 </dl>
 
+## Invalid Internal Objects &amp; Contagious Invalidity ## {#invalidity}
```
Comment (JusSn) [inline spec/index.bs]: Did you coin this term? Is there going to be more added later to explain it, or does it just refer to the `If the device that owns an object is lost, the object becomes invalid` line?

Comment (kainino0x) [reply spec/index.bs]: "Contagious Invalidity" is an update of the term we used to use "Contagious Internal Nullability" to match the current wording ("invalid" objects).

Yes, I'm planning to write more on the topic along and formalize it more.

**Comment by kainino0x** [comment]
> Preview looks good; so the final result should be that index.bs is only a list of headings and includes?

Yes, I think so.

File: spec/index.bs (hunk 3 above)
Comment (kainino0x) [reply spec/index.bs]: (It refers to the fact that if you use an invalid object, like a GPUBuffer, in the creation of a new object, like a GPUBindGroup, the new object is also invalid)

**Comment by kainino0x** [comment]
Removed `make online` since it no longer works with this patch.

**Comment by kainino0x** [comment]
I had a discussion with some Bikeshed folks.
- The Bikeshed API doesn't currently support multiple files (unless rendering from an http: link to the .bs source)
- It will support it soon

I think we can require our authors to download Bikeshed, at least for the moment.

**Comment by kainino0x** [comment]
@Kangz @toji @Jiawei-Shao @shaoboyan @haoxli @jdashg are you relying on `make online` in your current workflow? Or would you be okay if it were removed (hopefully only for a few weeks)?

**Comment by toji** [comment]
I've been using `make online` pretty extensively, mostly because my initial stab at installing bikeshed locally ran into some dependency issues I didn't want to bother debugging at the time. I'll take another crack at getting it working, though.

**Comment by Kangz** [comment]
I'm not using make online. However my bikeshed version won't update for some reason so maybe I'll need to use it at some point.

**Comment by kainino0x** [comment]
Perhaps we should hold off until make online can support this. I'll ping on that issue if I don't hear back soon.

**Comment by toji** [comment]
FWIW I finally got my bikeshed dependencies sorted, so there's at least one less user of `make online`. :)

**Comment by kainino0x** [comment]
Stale, though we still probably would like to do this sometime.

Happened to think of this because I'm working on autogenerating the TS types and realized this is likely to break https://github.com/darionco/bikeshed-to-ts. @darionco

**Pull request closed without merging**